src/SqlSquared.purs

```diff
@@ -70,11 +68,13 @@ decodeJsonQuery = traverse decodeJson <=< Sig.decodeJsonSqlQueryF
 decodeJsonModule ∷ J.Json → Either String SqlModule
 decodeJsonModule = traverse decodeJson <=< Sig.decodeJsonSqlModuleF
 
-arbitrarySqlOfSize ∷ Int → Gen.Gen Sql
-arbitrarySqlOfSize = anaM $ Sig.arbitrarySqlF EJ.arbitraryEJsonF
+genSql ∷ ∀ m. Gen.MonadGen m ⇒ MonadRec m ⇒ m Sql
+genSql = Gen.sized $ anaM (Sig.genSqlF EJ.arbitraryEJsonF)
 
-arbitrarySqlQueryOfSize ∷ Int → Gen.Gen SqlQuery
-arbitrarySqlQueryOfSize = traverse arbitrarySqlOfSize <=< Sig.arbitrarySqlQueryF
+genSqlQuery ∷ ∀ m. Gen.MonadGen m ⇒ MonadRec m ⇒ m SqlQuery
+genSqlQuery =
+  Gen.sized $ traverse (flip Gen.resize genSql <<< const) <=< Sig.genSqlQueryF
 
-arbitrarySqlModuleOfSize ∷ Int → Gen.Gen SqlModule
-arbitrarySqlModuleOfSize = traverse arbitrarySqlOfSize <=< Sig.arbitrarySqlModuleF
+genSqlModule ∷ ∀ m. Gen.MonadGen m ⇒ MonadRec m ⇒ m SqlModule
+genSqlModule =
+  Gen.sized $ traverse (flip Gen.resize genSql <<< const) <=< Sig.genSqlModuleF
```

Comment:
Will ```(_ `Gen.resize` genSql <<< const)``` work here?  (I find `_` easier to read than `flip`)

Reply:
You'd need some more parens `((_ `Gen.resize` genSql) <<< const)` which probably tips it into being worse than using `flip` again, I think. It's syntactically valid without the parens, but means `_ `Gen.Resize` (genSql <<< const)` if you don't include them, so it fails to typecheck instead... a little weird!